diff -urN uucp-1.07.org/configure.in uucp-1.07/configure.in
--- uucp-1.07.org/configure.in	2003-06-13 19:12:28.000000000 +0200
+++ uucp-1.07/configure.in	2003-06-13 19:21:25.000000000 +0200
@@ -361,8 +361,8 @@
 AC_CHECK_FUNC(getdelim,
 [AC_CHECK_FUNC(getline,
 [AC_DEFINE(HAVE_GETLINE, 1, [Define if you have the getline function.])],
-[LIBOBJS="$LIBOBJS getlin.o"])],
-[LIBOBJS="$LIBOBJS getlin.o"])
+[AC_LIBOBJ([getlin])])],
+[AC_LIBOBJ([getlin])])
 dnl
 dnl Check for the SCO buggy ftime; the code can cope with the bug,
 dnl though it would prefer not to, so if we're cross-configuring we
@@ -465,52 +465,51 @@
 dnl
 dnl Figure out which functions we need from lib subdirectory
 if test $ac_cv_func_bsearch != yes; then
-  LIBOBJS="$LIBOBJS bsrch.o"
+  AC_LIBOBJ([bsrch])
 fi
 if test $ac_cv_func_bzero != yes \
    && test $ac_cv_func_memset != yes; then
-  LIBOBJS="$LIBOBJS bzero.o"
+  AC_LIBOBJ([bzero])
 fi
 if test $ac_cv_func_memchr != yes; then
-  LIBOBJS="$LIBOBJS memchr.o"
+  AC_LIBOBJ([memchr])
 fi
 if test $ac_cv_func_memcmp != yes \
    && test $ac_cv_func_bcmp != yes; then
-  LIBOBJS="$LIBOBJS memcmp.o"
+  AC_LIBOBJ([memcmp])
 fi
 if test $ac_cv_func_memcpy != yes \
    && test $ac_cv_func_bcopy != yes; then
-  LIBOBJS="$LIBOBJS memcpy.o"
+  AC_LIBOBJ([memcpy])
 fi
 if test $ac_cv_func_strcasecmp != yes \
    && test $ac_cv_func_stricmp != yes; then
-  LIBOBJS="$LIBOBJS strcas.o"
+  AC_LIBOBJ([strcas])
 fi
 if test $ac_cv_func_strchr != yes \
    && test $ac_cv_func_index != yes; then
-  LIBOBJS="$LIBOBJS strchr.o"
+  AC_LIBOBJ([strchr])
 fi
 if test $ac_cv_func_strdup != yes; then
-  LIBOBJS="$LIBOBJS strdup.o"
+  AC_LIBOBJ([strdup])
 fi
 if test $ac_cv_func_strncasecmp != yes \
    && test $ac_cv_func_strnicmp != yes; then
-  LIBOBJS="$LIBOBJS strncs.o"
+  AC_LIBOBJ([strncs])
 fi
 if test $ac_cv_func_strrchr != yes \
    && test $ac_cv_func_rindex != yes; then
-  LIBOBJS="$LIBOBJS strrch.o"
+  AC_LIBOBJ([strrch])
 fi
 if test $ac_cv_func_strstr != yes; then
-  LIBOBJS="$LIBOBJS strstr.o"
+  AC_LIBOBJ([strstr])
 fi
 if test $ac_cv_func_strtol != yes; then
-  LIBOBJS="$LIBOBJS strtol.o"
+  AC_LIBOBJ([strtol])
 fi
 if test $ac_cv_func_strtoul != yes; then
-  LIBOBJS="$LIBOBJS strtou.o"
+  AC_LIBOBJ([strtou])
 fi
-AC_SUBST(LIBOBJS)dnl
 dnl Figure out which functions we need from unix subdirectory
 if test $ac_cv_func_opendir != yes; then
   UNIXOBJS="$UNIXOBJS dirent.o"
